pandas/io/json/_json.py

```diff
@@ -947,14 +950,18 @@ def read(self) -> DataFrame | Series:
             if self.engine == "pyarrow":
                 pyarrow_json = import_optional_dependency("pyarrow.json")
                 pa_table = pyarrow_json.read_json(self.data)
+
+                mapping: dict | type[ArrowDtype] | None
                 if self.dtype_backend == "pyarrow":
-                    return pa_table.to_pandas(types_mapper=ArrowDtype)
+                    mapping = ArrowDtype
                 elif self.dtype_backend == "numpy_nullable":
                     from pandas.io._util import _arrow_dtype_mapping
 
                     mapping = _arrow_dtype_mapping()
```

Comment:
Does this need to be `_arrow_dtype_mapping().get`?

Reply:
Yeah, not sure if we have a test that hits this though, will look in a follow up